### ISSUE TITLE
Improve Tab User Interface by merging into a single sheet and adding scroll to (anchor) links

### ISSUE THREAD

**Issue by bryancasler** [sdenec/tidy5e-sheet#97]
Hi all, I wanted to make a pitch for a usability improvement to the Character Sheets. Currently every section is a tab (e.g. Attributes, Inventory, etc..). This is a great structure but having to click the tab means a precise interaction of moving your cursor to that tab, clicking it, and then possibly even having to then scroll to the section within the tab you want.

As an alternative, I wanted to pitch making the character sheet into one long scrollable sheet. The tabs would still remain, and clicking them would jump you to the respective section of the sheet. In addition when you scroll to that respective section the tab would update to reflect that. With this change you can continue to use the sheet like you were with tabs, but now you can also take advantage of all the content being presented on one sheet and you can quickly scroll to where you would like to be.

Here's a screen recording showing what it would look like to have all the sections one after another with scrolling. It would need some styling clean up and possible section headers, but I think you can get a sense for how nice it would be to be able to have the option to quickly scrub/scroll through a character sheet without a majority of the content requiring multiple clicks to get to.

Screen Recording: https://d.pr/v/Eyey5d

Let me know what you think. I could help with some of the CSS styling and converting the tabs to anchor links. Probably would need help with the scroll position monitoring so the tabs auto-update as you scroll to them.

**Comment by sdenec** [comment]
Hi! Thanks for your input. I'm not huge fan of your suggestion though because of multiple things. That doesn't mean you can't fork the sheet and make your own design choices for your group if you wish to do so! But I won't change the current layout. Sorry. Here is why.

The separation in to tabs follows the base sheet, so no additional programming is required to change the sheet's behaviour - that is merely a point of laziness, granted. A continous sheet would certainly possible (as shown by you) - easily with a few css changes and some javascript adjustment.

But the separation into tabs makes it very clear where information can be found and it is neatly organized. Changing out tab links to jump markers is in my opinion a lot more confusing. Evertime you open the sheet you will have to click a marker to jump to a specific section. I bet you will end up clicking the labels equally often as it is now.

Furthermore at least my players tend to forget where stuff is located. Spell DC for example. Now i can say: "Click your spellbook tab and look a the bottom!". With a continous sheet that info gets lost somewhere in between the potentially very long spell list and the following biography. Directing players to that info would require a lot of scrolling and searching.

Same with coins and carrying capacity.

If there are cruitial items, spells, skills or feats that should be easily accessible there is the favorite function that combines a lot into the first tab, removing the need to switch to other tabs.

So sorry, but it's a no from my side. For now I won't touch the current layout of the sheet as it works very well.

TL;DR:
I really don't think the sheet would benefit from a one page layout. it gets more confusing regarding stuff to scroll through as contents can get very long. Important stuff can be put on the First tab. On a single page the jump markers would most likely be clicked as often as the tabs are now.

**Comment by bryancasler** [comment]
I appreciate you taking the time to respond and I've been using the 5e Tidy Sheet module so long I completely forgot the tab interface is a native one. I may try taking this on myself and if the players like I'll return to make a second pitch.
